### ISSUE TITLE
Are there plans to enable automatic capture of GPS track log?

### ISSUE THREAD

**Issue by shaneclarke** [Esri/Survey123Community#59]
Are there plans to enable automatic capture of GPS track log - without the user actually necessarily capturing a survey?

In many cases there is a need to capture not only the surveys but also where the users have been. Real-time is useful, but not always essential.
For example, the app that Esri built for the polio vaccinations in Nigeria a few years ago had this requirement. It was important to not only know who had been surveyed/vaccinated, but also to know where the workers had been. This could be both for security reasons as well as for tracking that the surveys were actually captured where they were supposed to have been captured.

**Comment by survey123** [comment]
Thanks for your comment Shane. We will keep this in the back log. At the moment, no explicit plans to get this done in the very short term.  But this is certainly part of the XLSForm spec and as such, it is in our radar to implement.
We will prioritize work based on feedback, so if others think this is important... add your comments and explain why it is important to you. 

Thanks again.

**Comment by shaneclarke** [comment]
Here is another reason: Our professional services are about to start on a project for anti-poaching in Africa. They are considering using Collector, partly because Collector allows automatic capture of a GPS tracklog. I think Survey123 may be better suited to the application, since it is easier to use and supports more powerful forms for the range in the field. However, ideally it needs this capability, as the project calls for a dashboard to see where the vehicles as well as the rangers are. And it would be really useful to see where both had been.
Another reason why I think Survey123 is a better fit is that this anti-poaching app is jsut the first app. The goal is to have multiple apps/surveys for the ranger to have access to.

**Comment by sirws** [comment]
I would like to add that it would be great to be able to have it tracking in the background as well.  Collector only performs location tracking when the app is actually running and the device is on.  It would be great to give the option of tracking that way or a more global tracking mechanism that allows tracking cross-surveys and while the device is not on and the Survey 123 is not active.

**Comment by marikavertzonis** [comment]
"while the device is not on and the Survey 123 is not active"... really?

Or when you say device not on - do you mean when the screen is asleep? And when Survey 123 is not active - do you mean the specific survey that you are completing is not active, but the app is running in the background? Otherwise your really describing a companion app or even a companion device (if the Survey 123 device is off).

I also would like to see some sort of GPS tracking in Survey123 but the questions I'm keen to hear opinions on, are about how it should work:
 - Do I get to configure different tracking options for each survey? Or just once for all surveys?
 - What are those options? 
        * Is it just an on/off option and track-logs are automatically saved as 1 per day (simple)
        * Is there also an option for tracking interval, either by time or distance (a little more complicated)
        * Instead of automatic saving all day, is there a start/stop button? Where would it be? In the Survey 123 app window? Or on the Survey form itself? Or both?

I could argue for all of these cases - but would suit most people?

**Comment by shaneclarke** [comment]
Yes, "the device is not on" means the screen is asleep/turned off.
And, yes, when the Survey123 app is not active, but is in the background.
Here are some thoughts:
The GPS tracking should be a property of the Survey123 app, and not of an individual survey.
The capability should only be enabled with a named user account (i.e. not for public surveys). This will allow associating a named user with a tracklog.
Initially, the options should be limited:
- turn GPS tracking on by default when the app is started
- specify frequency, by time. (I don't think you want to allow specifying frequency by distance, as it is important to know how long the user spent at a specific location)
- option to specify how frequently points get sent to the server when in connected mode. (In some cases, it is important to send each GPS point when collected, while in other cases the points can be buffered and only sent in batches)
- turn off (or pause) the GPS tracking.
- a new tracklog should be created for each day, and named accordingly

The user should not have an option of clearing the track log - this will automatically happen when synced.

Hope this helps!

**Comment by kengorton** [comment]
Bump one with a major qualification. I believe this capability is crucial for the ArcGIS platform but should not necessarily be integrated into Survey123. A separate app could (and IMHO, should) do this because:
•	The feature schema of the tracklog would require a separate feature layer schema than that of the survey
•	this functionality could be used in many, many workflows within the platform including real-time workflows
•	The more features you add to Survey123, the more complicated it becomes for the user and we risk losing the "drop-dead easy" characteristic
•	Not all Survey123 collection platforms will have GPS capability
I believe we should build a separate app which will run in the background on GPS-enabled mobile devices collecting GPS points at a user-defined interval including, at an absolute minimum, time, longitude, latitude, altitude and device ID. It might also capture speed and direction of travel, battery level, GPS accuracy or other values exposed by the device's sensors.
This app should feed a separate service which could be a feature service in AGOL or ArcGIS Server or even a GeoEvent input. The app should detect the device’s connection status and:
•	If disconnected, store the points internally until such time as it is connected
•	If connected, offer a user setting to upload points in real time or log them on the device until such time as the user chooses to upload them, to favor a wifi connection over cellular data consumption. 
Survey123 might detect the presence of this GPS beacon app, and perhaps include a menu option to initiate tracking and even open the beacon app to configure tracking settings.

Just my two cents.
Ken

**Comment by broecher** [comment]
I have been using CyberTracker freeware for surveys on windows mobile and android for several years and it does have the ability to simultaneously take a track log. That app has a lot of GPS control that I would like to see in Survey123.

In cases where you need to record many records quickly, CT is faster since it uses a screen for each question with a next button instead of scrolling. When a screen needs a number the number keypad is part of the screen and already displayed for you. Select-one screens can be configured to move to the next screen without pressing next as soon as you make the selection.

A major benefit in CT is that taking a location is a property of the save button instead of a question. When the survey is configured to take a track log, the GPS is already 'hot' when you press save and an accurate location is saved immediately. When the track log is not on, the GPS is 'cold' and it takes longer to get the fix and the accuracy is occasionally really bad.

I suggest that Survey123 developers take a look at CT, as they did with ODK.

**Comment by survey123** [comment]
Thanks a lot for your feedback. We really appreciate your comments! We are definitively thinking along your lines and will try to improve things  one at a time.
